vulnerabilities/importers/github.py

```diff
@@ -146,189 +122,339 @@
+
+class GitHubBasicImprover(Improver):
+    @property
+    def interesting_advisories(self) -> QuerySet:
+        return Advisory.objects.filter(created_by=GitHubAPIImporter.qualified_name)
+
+    def get_inferences(self, advisory_data: AdvisoryData) -> Iterable[Inference]:
+        """
+        Generate and return Inferences for the given advisory data
+        """
+        try:
+            purl, affected_version_ranges, _ = AffectedPackage.merge(
+                advisory_data.affected_packages
+            )
+        except KeyError:
+            return iter([])
+        pkg_name = purl.name
+        pkg_namespace = purl.namespace
+        pkg_type = purl.type
+        ecosystem = get_ecosystem_from_type(pkg_type)
+        self.set_api(ecosystem, name)
+        name = process_name_namespace(ecosystem, pkg_name, pkg_namespace)
+        if hasattr(self.version_api, "module_name_by_package_name"):
+            pkg_name = self.version_api.module_name_by_package_name.get(name, pkg_name)
+        for affected_version_range in affected_version_ranges:
+            aff_vers, unaff_vers = categorize_versions(
+                ecosystem,
+                affected_version_range,
+                self.version_api.get(name, until=advisory_data.date_published).valid_versions,
+            )
+            affected_purls = [
+                PackageURL(name=pkg_name, namespace=pkg_namespace, version=version, type=pkg_type)
+                for version in aff_vers
+            ]
+
+            unaffected_purls = [
+                PackageURL(
+                    name=purl.name, namespace=purl.namespace, version=version, type=purl.type
+                )
+                for version in unaff_vers
+            ]
+
+            for fixed_purl in unaffected_purls:
+                yield Inference.from_advisory_data(
+                    advisory_data,
+                    confidence=90,
+                    affected_purls=affected_purls,
+                    fixed_purl=fixed_purl,
+                )
+
+    def set_api(self, ecosystem):
+        self.version_api = get_version_api(ecosystem)
+
+
+def categorize_versions(
+    ecosystem: str, version_range: VersionRange, all_versions: Set[str]
+) -> Tuple[List[str], List[str]]:
+    """
+    Return `affected_versions` and `unaffected_versions`
+    iterate on `all_versions` and if the version is affected by `version_range`
+    append it in `affected_versions`
+    else append it in `unaffected_versions`
+    """
+    version_class: VersionRange = ECOSYSTEM_VERSION_RANGE_MAPPING.get(ecosystem)
+    affected_versions = []
+    unaffected_versions = []
+    for version in all_versions:
+        if version in IGNORE_VERSIONS:
+            continue
+
+        if version in version_class.from_string(version_range.to_string()):
+            affected_versions.append(version)
+        else:
+            unaffected_versions.append(version)
+    return (affected_versions, unaffected_versions)
```

Comment:
nit: no parens needed
```suggestion
    return affected_versions, unaffected_versions
```